client/blocks/app-banner/index.jsx

```diff
@@ -65,6 +66,15 @@ class AppBanner extends Component {
 		userAgent: ( typeof window !== 'undefined' ) ? navigator.userAgent : '',
 	};
 
+	preventNotificationsClose( appBanner ) {
+		const appBannerNode = ReactDom.findDOMNode( appBanner );
+		if ( ! appBannerNode ) {
+			return;
+		}
+		appBannerNode.addEventListener( 'mousedown', event => event.stopPropagation(), false );
+		appBannerNode.addEventListener( 'touchstart', event => event.stopPropagation(), false );
+	}
```

Comment:
We should get a `null` value here when/if the `<Card />` unmounts. Honestly I'm not sure if we get that when the parent unmounts though. When working with the DOM shouldn't we be careful not to open up memory leaks by attaching event listeners but not removing them? Also, I think that `appBanner` in this case is the _actual DOM Node_ and we don't need `findDOMNode()`…not confident on that mainly because we're attaching a `ref` to an `Element` and not a `DOMNode` directly

Maybe instead we could rearrange this based on the `null`ity of `appBanner` the ref…

```js
eventStopper = event => event.stopPropagation();

preventNotificationsClose( appBanner ) {
	if ( ! appBanner && this.appBanner ) {
		this.appBanner.removeEventListener( 'mousedown', eventStopper, false );
		this.appBanner.removeEventListener( 'touchstart', eventStopper, false );
		return;
	}

	if ( appBanner ) {
		this.appBanner = appBanner;
		appBanner.addEventListener( … );
		appBanner.addEventListener( … );
		return;
	}
}
```

anyway, I know DOM stuff requires more attention and I just want to make sure we cleanup